src/_pytest/terminal.py

```diff
@@ -1022,6 +1024,20 @@ def summary_passes(self) -> None:
                         self._outrep_summary(rep)
                     self._handle_teardown_sections(rep.nodeid)
 
+    def summary_xpasses(self) -> None:
+        if self.config.option.tbstyle != "no":
+            if self.hasopt("X"):
```

Comment:
I'm curious, do we need this `if`? I mean, `self.getreports("xpassed")` in the line below will report an empty list if `-rX` was not passed anyway, correct?

Reply:
I think we need it.
I tried removing it, and then the xfail reports always get printed.

Reply:
Ahh of course that's right: the reports are always reported, the flag controls if we should display them.